### ISSUE TITLE
How to detect scrolling on CameraControls

### ISSUE THREAD

**Issue by IDrumsey** [yomotsu/camera-controls#436]
As stated [here](https://yomotsu.github.io/camera-controls/#:~:text=mouseButtons.wheel%20(Mouse%20wheel%20control)%20does%20not%20emit%20%27controlstart%27%20and%20%27controlend%27.%20mouseButtons.wheel%20uses%20scroll%2Devent%20internally%2C%20and%20scroll%2Devent%20happens%20intermittently.%20That%20means%20%22start%22%20and%20%22end%22%20cannot%20be%20detected.)

> mouseButtons.wheel (Mouse wheel control) does not emit 'controlstart' and 'controlend'. mouseButtons.wheel uses scroll-event internally, and scroll-event happens intermittently. That means "start" and "end" cannot be detected.

Is there any way to detect the scroll event on the CameraControls? I've tried putting the event listener 'scroll' on the Canvas and a on a wrapper div around the Canvas. Nothing gets fired. My use case is when the user has scrolled to the minDistance or maxDistance, the zooming will stop and the mouse wheel will start to regularly scroll that page.

**Comment by yomotsu** [comment]
Unfortunately, CC will block scroll currently... 

CC prevents wheel events, then scroll events will be blocked as a side effect.
here is the code: https://github.com/yomotsu/camera-controls/blob/cee042753169f3bbeb593833ce92d70d52b6862f/src/CameraControls.ts#L840-L866

Maybe we could add a new option like `releaseWheelOnEdges` to disable the event prevention for wheel events.
(any idea for good naming for the new option...?)

**Comment by IDrumsey** [comment]
@yomotsu Is zooming with touch events done differently than scrolling the page with touch events? Just thinking... on desktop you zoom with the mouse wheel AND page scroll with the mouse wheel. But if on mobile zooming is done with two fingers and scrolling is performed with a different action, we might not need to include touch events for this new feature.

Some naming options...

- normalScrollOnZoomBounds
- scrollOnZoomBounds
- swapZoomToScrollOnZoomBounds

just a few ideas

**Comment by esnho** [comment]
@IDrumsey on mobile the scrolling is prevented by the camera orbit event, would be great if CC let the page scroll when the orbit reached its limit!

@yomotsu what do you think of an option like `releaseEventOnEdges` available on those behaviours that can reach an edge?
Probably on scrolling is something easier than on dragging

**Comment by IDrumsey** [comment]
@esnho, yeah. I forgot about orbiting constraints. Would make sense to start auto scrolling  when the limits were reached.

Not familiar with the codebase, but maybe something like `releaseEventOnEdges(eventTypes: [Zoom, Orbit] = [all the events]){}`

**Comment by yomotsu** [comment]
Thank you, guys, for your suggestions.
The name `releaseEventOnEdges` sounds to cover both wheel and touch events.
Let me use your name!

It may take time, but I will add features when I have time.

**Comment by SrinivasPrabhu794** [comment]
Hi,

Will this be released anytime soon ?

**Comment by omalige** [comment]
Hi, I'm bringing this up again because adding this property would be very useful, are there any plans to add it?
